src/MasterBlueprintServiceManager.sol

```diff
@@ -0,0 +1,482 @@
+    }
+
+    /**
+     * @dev Called when a service request is rejected by an operator.
+     * @param blueprintId The blueprint unique identifier.
+     * @param operator The operator's details.
+     * @param requestId The ID of the request.
+     */
+    function onReject(
+        uint64 blueprintId,
+        ServiceOperators.OperatorPreferences calldata operator,
+        uint64 requestId
+    )
+        public
+        onlyFromRootChain
+        whenNotPaused
+    {
+        address manager = blueprints.get(blueprintId);
+        IBlueprintServiceManager(manager).onReject(operator, requestId);
+        emit RequestRejected(blueprintId, requestId, operator);
+    }
+
+    /**
+     * @dev Called when a service is initialized.
+     * @param blueprintId The blueprint unique identifier.
+     * @param requestId The ID of the request.
+     * @param serviceId The ID of the service.
+     * @param owner The owner of the service.
+     * @param permittedCallers The list of permitted callers.
+     * @param ttl The time-to-live for the service.
+     */
+    function onServiceInitialized(
+        uint64 blueprintId,
+        uint64 requestId,
+        uint64 serviceId,
+        address owner,
+        address[] calldata permittedCallers,
+        uint64 ttl
+    )
+        public
+        onlyFromRootChain
+        whenNotPaused
+    {
+        address manager = blueprints.get(blueprintId);
+        IBlueprintServiceManager(manager).onServiceInitialized(requestId, serviceId, owner, permittedCallers, ttl);
+        emit ServiceInitialized(blueprintId, requestId, serviceId, owner, ttl);
+    }
+
+    /**
+     * @dev Called when a job is called within the service context.
+     * @param blueprintId The blueprint unique identifier.
+     * @param serviceId The ID of the service.
+     * @param job The job identifier.
+     * @param jobCallId The unique ID for the job call.
+     * @param inputs The inputs required for the job execution.
+     */
+    function onJobCall(
+        uint64 blueprintId,
+        uint64 serviceId,
+        uint8 job,
+        uint64 jobCallId,
+        bytes calldata inputs
+    )
+        public
+        payable
+        onlyFromRootChain
+        whenNotPaused
+    {
+        address manager = blueprints.get(blueprintId);
+        IBlueprintServiceManager(manager).onJobCall(serviceId, job, jobCallId, inputs);
+        emit JobCalled(blueprintId, serviceId, job, jobCallId);
+    }
+
+    /**
+     * @dev Called when operators send the result of a job execution.
+     * @param blueprintId The blueprint unique identifier.
+     * @param serviceId The ID of the service.
+     * @param job The job identifier.
+     * @param jobCallId The unique ID for the job call.
+     * @param operator The operator sending the result.
+     * @param inputs The inputs used for the job execution.
+     * @param outputs The outputs from the job execution.
+     */
+    function onJobResult(
```

Comment:
In the spirit of knocking it all out, we would also benefit from `onResultSubmitted` or `onJobResultSubmitted` so that it's very clear, wdyt?

Reply:
It is analogous to `onJobCall` so we have `onJobResult`. Naming here is not a big issue, but I will go with obvious naming here.